tests/test_gapfit.py

```diff
@@ -66,20 +82,37 @@ def check_gap_fit(self, command_line, ref_data, new_test=False):
         else:
             print('max abs error in alpha =', np.abs((alpha - ref_data['alpha'])).max())
             assert np.abs((alpha - ref_data['alpha'])).max() < 1e-2
-                
+
     def test_gap_fit_silicon(self):
         train_filename = 'train_sub4.xyz'
         command_line = (f"at_file={train_filename} gap={{soap l_max=8 n_max=8 atom_sigma=0.5 "
                         "zeta=4 cutoff=4.0 cutoff_transition_width=1.0 central_weight=1.0 "
                         "n_sparse=500 delta=3.0 f0=0.0 covariance_type=dot_product "
-                        "SPARSE_METHOD } "
+                        "SPARSE_METHOD} "
                         "default_sigma={0.001 0.1 0.05 0.0} "
                         "energy_parameter_name=dft_energy force_parameter_name=dft_force "
                         "virial_parameter_name=dft_virial config_type_parameter_name=config_type "
-                        "sparse_jitter=1.0e-8 e0_offset=2.0 gp_file=gp.xml rnd_seed=1")
+                        "sparse_jitter=1.0e-8 e0_offset=2.0 gp_file=gp.xml rnd_seed=1 do_condition_number=T")
         with open('si_gap_fit_test.json') as f:
             ref_data = json.load(f)
         self.check_gap_fit(command_line, ref_data, new_test=False)
 
+    @unittest.skipIf(os.environ.get('HAVE_SCALAPACK') != '1', 'ScaLAPACK support not enabled')
+    def test_gap_fit_silicon_scalapack(self):
+        command_line = string.Template("at_file=train_sub4.xyz gap={soap l_max=8 n_max=8 atom_sigma=0.5 "
+                        "zeta=4 cutoff=4.0 cutoff_transition_width=1.0 central_weight=1.0 "
+                        "n_sparse=500 delta=3.0 f0=0.0 covariance_type=dot_product "
+                        "$SPARSE_METHOD} "
+                        "default_sigma={0.001 0.1 0.05 0.0} "
+                        "energy_parameter_name=dft_energy force_parameter_name=dft_force "
+                        "virial_parameter_name=dft_virial config_type_parameter_name=config_type "
+                        "sparse_jitter=1.0e-8 e0_offset=2.0 gp_file=gp.xml rnd_seed=1")
+        with open('si_gap_fit_test.json') as f:
+            ref_data = json.load(f)
+
+        self.check_gap_fit(command_line.substitute(SPARSE_METHOD='sparse_method=cur_points'), ref_data, new_test=False, prefix='mpirun -np 1')
```

Comment:
to save time and increase robustness we could store the commit the set of sparse points to the repo - I think this what I did for the serial test?

Reply:
Thought about it. Wasn't sure if we wanted to fill the the repo with dummy files.

Reply:
If there's on there already for the serial test can't. we use the same sparse points?

Reply:
I didn't think there was a sparse points file. I will have a look.

Reply:
Looks like the serial test just reuses a sparse index file. That does not work for MP as the processes see only part of the data.

Reply:
So I will remove the np1 test and add sparseX.inp to the repo. Anything else?

Reply:
> Looks like the serial test just reuses a sparse index file. That does not work for MP as the processes see only part of the data.

I don't think that's correct- all processes have all sparse points (M), it's the data direction of the matrix (N) which is distributed.

Reply:
But where do they get the sparse coords from? I thought they process all the frames in xyz, then pick via CUR or index_point file which points they use for M. In the MP case each worker reads all the frames only to count the numbers, then throws that away and reads only the frames they are interested in. If so, the sparse_index file should not be enough to get the same coords on all workers.

Reply:
I tested this and the results were way off. I have checked in the sparseX.inp file now. We should keep in mind that any file we add will stay in the repo even if we change the test later.

Reply:
An alternative would be to depend on the serial test to provide the sparseX file. I did not find the right unittest calls for that and it is against the spirit of independent unit tests so maybe it is not as available as I hoped when I looked for it.